### PR TITLE
[Playwright] PEPPER-308 Run playwright tests when PR merge into develop branch

### PR DESCRIPTION
**Pull request by aweng98**
Reworked `build-and-deploy-all-apps-workflow` workflow to allow running Playwright tests after deploys have been finished. Running build, unit tests and deploys are conditional for a study, happenning only if the study has changed.

New workflow name is `build-test-deploy-apps`. The workflow diagram will look similar to this screenshot

Screenshot 2023-02-23 at 6 55 32 PM

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -295,44 +291,31 @@ commands:
             ng build $ANGULAR_PROJECT_NAME $BUILD_PARAMS --configuration production --aot --base-href=/ --output-path=$OUTPUT_DIR --verbose=true --progress=true
           #
 
-  conditionally-launch-build-and-deploy:
-    description: Check every study, launch build and deploy if study has changed
+  halt-when-study-unchanged:
```
Comment (aweng98) [inline .circleci/config.yml:303]: `app-build`, `app-deploy` and `ui-unit-test` jobs will have this optional check to determine whether job should stop if study has not changed.

File: .circleci/config.yml [hunk 2]
```diff
@@ -412,28 +395,24 @@ jobs:
...
+            - halt-when-study-unchanged:
+                study_key: << parameters.study_key >>
+      - set-deployment-environment
+      - setup-shared-env:
           study_key: << parameters.study_key >>
-      - run-api-unit-test
       - deploy-exploded-build:
           study_key: << parameters.study_key >>
 
-  conditionally-launch-build-and-deploy-all-job:
```
Comment (aweng98) [inline .circleci/config.yml:426]: No longer used.

File: .circleci/config.yml [hunk 3]
```diff
@@ -793,31 +789,54 @@ workflows:
...
-        - not: << pipeline.parameters.do-builds >>
-        - not: << pipeline.parameters.scheduled-playwright >>
+  build-test-deploy-apps:
+    # conditionally build and deploy for changed studies on develop branch
     jobs:
-      - conditionally-launch-build-and-deploy-all-job:
+      - app-build:
           <<: *filter-develop-branch
-
-  build-and-deploy-workflow:
```
Comment (aweng98) [inline .circleci/config.yml:805]: `build-and-deploy-workflow` workflow is no longer used. See comment above. It was triggerd by CI api call from inside the`build-and-deploy-all-apps-workflow` workflow job.

File: .circleci/config.yml [hunk 4]
```diff
@@ -793,31 +789,54 @@ workflows:
           requires:
             - playwright-build
 
-  build-and-deploy-all-apps-workflow:
```
Comment (aweng98) [inline .circleci/config.yml:796]: Old workflow diagram looks similar to this screenshot. `app-deploy` for all studies are separate workflows triggered by `build-and-deploy-all-apps-workflow` workflow. 

Screenshot 2023-02-23 at 5 48 20 PM

**Review by Rinchan20 — APPROVED** [review]

**Pull request closed without merging**